### PR TITLE
feat: implement `useRenderCount` hook

### PR DESCRIPTION
**Pull request by ChloeMouret**
## What new hook does?

useRendersCount tracks component's renders count including the first render.

#33 

## Checklist

- [X] Have you read [contribution guideline](../../CONTRIBUTING.md)?
- [X] Have the files been linted and formatted?
- [X] Have the docs been updated?
- [X] Have the tests been added to cover new hook?
- [X] Have you run the tests locally to confirm they pass?

### CODE REVIEW

File: src/useRendersCount/__docs__/example.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import React, { useState } from 'react';
+import { useRendersCount } from '../useRendersCount';
+
+export const Example: React.FC = () => {
+  const [count, setCount] = useState(0);
```
Comment (xobotyi) [inline src/useRendersCount/__docs__/example.stories.tsx:5]: Better to use `useRerender` hook.

In our examples we're trying to reuse our hooks as much as possible.

**Review by xobotyi — CHANGES_REQUESTED** [review]
1) You havent updated `README.md`
2) `index.ts` is missing reexport of this hook
3) You should add note within [migrating-from-react-use.story.mdx](https://github.com/react-hookz/web/blob/master/src/__docs__/migrating-from-react-use.story.mdx)

File: src/useRendersCount/__tests__/dom.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import { useRendersCount } from '../useRendersCount';
+
+describe('useRendersCount', () => {
+  it('should be defined', () => {
+    expect(useRendersCount).toBeDefined();
+  });
+
+  it('should return number', () => {
+    expect(renderHook(() => useRendersCount()).result.current).toEqual(expect.any(Number));
+  });
```
Comment (ArttuOll) [inline src/useRendersCount/__tests__/dom.ts:11]: This test is unnecessary, since TypeScript guarantees that a number is returned.

**Review by ArttuOll — APPROVED** [review]
Pushed a couple of typo fixes, otherwise looks good to me!

**Pull request merged**